api/python/cellxgene_census/src/cellxgene_census/experimental/ml/pytorch.py

```diff
@@ -196,16 +196,38 @@ def obs_batch(self) -> pd.DataFrame:
         self.stats.nnz += self.X_batch.nnz
         self.stats.elapsed += int(time() - start_time)
         self.stats.n_soma_batches += 1
+        # TODO: also show per-batch stats (non-cumulative)
         pytorch_logger.debug(f"Retrieved batch: shape={self.X_batch.shape}, cum_stats: {self.stats}")
         return self.obs_batch_
 
 
 class ExperimentDataPipe(pipes.IterDataPipe[Dataset[ObsDatum]]):  # type: ignore
     """
-    An iterable-style data pipe.
+    An iterable-style PyTorch data pipe that reads obs and X data from a SOMA Experiment, and returns an iterator of
```

Comment:
I know you consider them redundant, but I still think these need lifecycle tags.  As Brian said, "belt and suspenders"